src/server/components/delete-element-button.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { OutlineButton, TextButton } from './buttons'
+import Modal from './modal'
+import ModalTitle from './modal-title'
+import ModalContent from './modal-content'
+import ModalCall from './modal-call'
+
+
+function getCookie(name) {
+  if (!document.cookie) {
+    return null
+  }
+  const token = document.cookie.split(';')
+    .map(c => c.trim())
+    .filter(c => c.startsWith(`${name}=`))
+
+  if (token.length === 0) {
+    return null;
+  }
+  return decodeURIComponent(token[0].split('=')[1])
+}
+
+export default class DeleteObjectButton extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { visible: false }
+    this.hideModal = this.hideModal.bind(this)
+    this.showModal = this.showModal.bind(this)
+    this.deleteObject = this.deleteObject.bind(this)
+  }
+
+  hideModal() {
+    this.setState({ visible: false })
+  }
+
+  showModal() {
+    this.setState({ visible: true })
+  }
+
+  deleteObject() {
+    const csrfToken = getCookie('csrftoken')
+    const headers = {
+      'Content-Type': 'application/json',
+      'X-CSRFToken': csrfToken,
+    };
```

Comment:
This is the second time I think we're constructing these headers. I feel like we should abstract away this csrf stuff somehow, either with our own helper library/code or something from npm.

Reply:
Yep, good point. I'll abstract it out as a util fcn that we can reuse elsewhere.